src/backend/src/user_profile.rs

```diff
@@ -35,3 +40,31 @@ pub fn create_profile(
         default_profile
     }
 }
+
+pub fn add_credential(
+    principal: StoredPrincipal,
+    credential_type: &CredentialType,
+    user_profile_map: &mut StableBTreeMap<(u64, StoredPrincipal), Candid<StoredUserProfile>, VMem>,
+    user_profile_updated_map: &mut StableBTreeMap<StoredPrincipal, u64, VMem>,
+) -> Result<(), AddUserCredentialError> {
+    if let Ok(user_profile) = get_profile(principal, user_profile_map, user_profile_updated_map) {
+        let now = time();
+        let user_credential = UserCredential {
+            credential_type: credential_type.clone(),
+            verified_date_timestamp: Some(now),
+        };
+        let mut credentials = user_profile.credentials.clone();
```

Comment:
Not sure, should we set the `user_profile` as mutable to direcly modif the credentials and spare this clone?

Reply:
Discussed offline. I implemented the TokenVersion and therefore cloned a whole user_profile and credentials.